test/e2e/auth/audit.go

```diff
@@ -23,83 +23,623 @@ import (
+						v1beta1.StageResponseComplete,
+						fmt.Sprintf("/api/v1/namespaces/%s/pods", namespace),
+						"create",
+						201,
+						auditTestUser,
+						"pods",
+						namespace,
+						true,
+						true,
+					}, {
+						v1beta1.LevelRequest,
+						v1beta1.StageResponseComplete,
+						fmt.Sprintf("/api/v1/namespaces/%s/pods/audit-pod", namespace),
+						"get",
+						200,
+						auditTestUser,
+						"pods",
+						namespace,
+						false,
+						false,
+					}, {
+						v1beta1.LevelRequest,
+						v1beta1.StageResponseComplete,
+						fmt.Sprintf("/api/v1/namespaces/%s/pods", namespace),
+						"list",
+						200,
+						auditTestUser,
+						"pods",
+						namespace,
+						false,
+						false,
+					}, {
+						v1beta1.LevelRequest,
+						v1beta1.StageResponseStarted,
+						fmt.Sprintf("/api/v1/namespaces/%s/pods?timeoutSeconds=%d&watch=true", namespace, watchTestTimeout),
+						"watch",
+						200,
+						auditTestUser,
+						"pods",
+						namespace,
+						false,
+						false,
+					}, {
+						v1beta1.LevelRequest,
+						v1beta1.StageResponseComplete,
+						fmt.Sprintf("/api/v1/namespaces/%s/pods?timeoutSeconds=%d&watch=true", namespace, watchTestTimeout),
+						"watch",
+						200,
+						auditTestUser,
+						"pods",
+						namespace,
+						false,
+						false,
+					}, {
+						v1beta1.LevelRequestResponse,
+						v1beta1.StageResponseComplete,
+						fmt.Sprintf("/api/v1/namespaces/%s/pods/audit-pod", namespace),
+						"update",
+						200,
+						auditTestUser,
+						"pods",
+						namespace,
+						true,
+						true,
+					}, {
+						v1beta1.LevelRequestResponse,
+						v1beta1.StageResponseComplete,
+						fmt.Sprintf("/api/v1/namespaces/%s/pods/audit-pod", namespace),
+						"patch",
+						200,
+						auditTestUser,
+						"pods",
+						namespace,
+						true,
+						true,
+					}, {
+						v1beta1.LevelRequestResponse,
+						v1beta1.StageResponseComplete,
+						fmt.Sprintf("/api/v1/namespaces/%s/pods/audit-pod", namespace),
+						"delete",
+						200,
+						auditTestUser,
+						"pods",
+						namespace,
+						true,
+						true,
+					},
+				},
+			}, {
+				"Deployment",
+				func() {
+					podLabels := map[string]string{"name": "audit-deployment-pod"}
+					d := framework.NewDeployment("audit-deployment", int32(1), podLabels, "redis", imageutils.GetE2EImage(imageutils.Redis), extensions.RecreateDeploymentStrategyType)
+
+					_, err := f.ClientSet.Extensions().Deployments(f.Namespace.Name).Create(d)
+					framework.ExpectNoError(err, "failed to create audit-deployment")
+
+					_, err = f.ClientSet.Extensions().Deployments(f.Namespace.Name).Get(d.Name, metav1.GetOptions{})
+					framework.ExpectNoError(err, "failed to get audit-deployment")
+
+					deploymentChan, err := f.ClientSet.Extensions().Deployments(f.Namespace.Name).Watch(watchOptions)
+					for range deploymentChan.ResultChan() {
+					}
+					framework.ExpectNoError(err, "failed to create watch for deployments")
```

Comment:
Same as above: error should be before waiting for channel

Reply:
Done.